### PR TITLE
[sw/silicon_creator] Enable jittery clock if enabled in OTP

### PR DESCRIPTION
**Pull request by alphan**
~@tjaychen @moidx  I ended up checking the value of the OTP entry instead of writing directly to the register because according to the spec:~

> A value of kMultiBitBool4False disables the jittery clock, while all other values enable jittery clock.

Updated the change to write the OTP value to the clkmgr register.

Please also note that jittery clock is not enabled if AST initialization is skipped.

### CODE REVIEW

**Comment by tjaychen** [comment]
> @tjaychen @moidx I ended up checking the value of the OTP entry instead of writing directly to the register because according to the spec:
> 
> > A value of kMultiBitBool4False disables the jittery clock, while all other values enable jittery clock.
> 
> Please also note that jittery clock is not enabled if AST initialization is skipped.

@alphan actually turns out this statement in the spec is no longer true.
The way Nuvoton is consuming this signal is exactly the opposite. I will update the specification.

**Comment by alphan** [comment]
> > @tjaychen @moidx I ended up checking the value of the OTP entry instead of writing directly to the register because according to the spec:
> > > A value of kMultiBitBool4False disables the jittery clock, while all other values enable jittery clock.
> > 
> > 
> > Please also note that jittery clock is not enabled if AST initialization is skipped.
> 
> @alphan actually turns out this statement in the spec is no longer true. The way Nuvoton is consuming this signal is exactly the opposite. I will update the specification.

done

**Review by moidx — APPROVED** [review]
LG. Does a device in TEST LC skip AST INIT as well?

**Comment by alphan** [comment]
@tjaychen reverted the last change as we discussed, ptal.

File: sw/device/silicon_creator/mask_rom/mask_rom_start.S [hunk 1]
```diff
@@ -185,6 +186,17 @@ _mask_rom_start_boot:
   andi t0, t0, 0x1
   beqz t0, .L_ast_done_loop
 
+  // Check if jittery clock should be enabled.
+  li   a0, (TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR + \
+            OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET)
+  lw   t0, OTP_CTRL_PARAM_CREATOR_SW_CFG_JITTER_EN_OFFSET(a0)
```
Comment (tjaychen) [inline sw/device/silicon_creator/mask_rom/mask_rom_start.S:192]: @alphan i think you might want to do the same thing as what the proposed HW change is doing.
Ie, if it's NOT false, then just copy over.  That will make sure even a glitch on the read, unless it glitches to false exactly.

**Comment by tjaychen** [comment]
just a note on rationale.
@alphan and i discussed a bit how to treat the jitter enable OTP.

My thinking was that since the jitter enable is bound to AST init, we can just directly copy from the OTP space without checking. 
That way if the OTP read value were somehow glitched to not-false, it would always turn on jitter. 

This brings up a question of what a raw otp value of 0 should do.
My thinking is that since it's bound to AST init, then we can simply decide what to do when we burn the OTP bits for AST init. We can always make it false and just let later stages of software change it.

Let me know if this sounds off to anyone.

**Comment by msfschaffner** [comment]
> just a note on rationale. @alphan and i discussed a bit how to treat the jitter enable OTP.
> 
> My thinking was that since the jitter enable is bound to AST init, we can just directly copy from the OTP space without checking. That way if the OTP read value were somehow glitched to not-false, it would always turn on jitter.
> 
> This brings up a question of what a raw otp value of 0 should do. My thinking is that since it's bound to AST init, then we can simply decide what to do when we burn the OTP bits for AST init. We can always make it false and just let later stages of software change it.
> 
> Let me know if this sounds off to anyone.

This sounds good to me!

**Review by tjaychen — APPROVED** [review]

**Pull request merged**